worlds/psychonauts/Rules.py

```diff
@@ -0,0 +1,506 @@
+            # BB Grindrail Wall requires Levitation because the ground around the cobweb is sloped and Raz bounces off
+            # it when falling onto it too fast, so Levitation is needed to float down slowly.
+            add_rule(multiworld.get_location(LocationName.CobwebGrindrailWall, player), self.has_levitation)
+
+        if local_only_forbidden:
+            def forbid_local_only(item: Item):
+                return item.player != player or item.name not in LOCAL_SET
+
+            for location_name in local_only_forbidden:
+                location = multiworld.get_location(location_name, player)
+                add_item_rule(location, forbid_local_only)
+
+    def set_psy_goal(self):
+        goal_option = self.world.options.Goal
+
+        # Goals are always logically completed in sequence:
+        # 1) Brain Hunt
+        # 2) Brain Tank
+        # 3) Meat Circus
+        # Victory is achieved upon completing the last goal in the sequence that is enabled.
+
+        # Either the Brain Hunt goal or the Brain Tank goal will always be enabled.
+
+        # Brain Hunt goal
+        if goal_option == Goal.option_brain_hunt or goal_option == Goal.option_asylum_brain_tank_and_brain_hunt:
+            redeemed_required_brains = self.multiworld.get_location(LocationName.RedeemedBrainsEvent, self.player)
+            brains_required = self.world.options.BrainsRequired.value
+            redeemed_required_brains.access_rule = lambda state: self.redeemed_brain_goal(state, brains_required)
+        else:
+            redeemed_required_brains = None
+
+        # Brain Tank goal
+        if goal_option == Goal.option_asylum_brain_tank or goal_option == Goal.option_asylum_brain_tank_and_brain_hunt:
+            oleander_boss_location = self.multiworld.get_location(LocationName.OleanderBrainTankBossEvent, self.player)
+            oleander_boss_location.access_rule = self.has_oleander_boss_access
+            # With Brain Hunt also enabled, accessing the Brain Tank is set to logically require completing the Brain
+            # Hunt first, though this is not currently enforced by Psychonauts.
+            if redeemed_required_brains:
+                add_rule(oleander_boss_location, redeemed_required_brains.can_reach)
+        else:
+            oleander_boss_location = None
+
+        # Meat Circus Final Boss goal
+        if self.world.options.RequireMeatCircus:
+            # The Meat Circus Final Boss can only be accessed after completing the previous enabled goal(s). This is
+            # enforced by Psychonauts.
+            final_boss_location = self.multiworld.get_location(LocationName.MeatCircusFinalBossEvent, self.player)
+            # If enabled, the Brain Tank will be the previous goal, otherwise it will be the Brain Hunt.
+            previous_goal_location = oleander_boss_location or redeemed_required_brains
+            final_boss_location.access_rule = previous_goal_location.can_reach
```

Comment:
Since this can't be `None`, you can assert it here (the if/else tree does not ensure one is hit without outside knowledge of the goal option values)
```suggestion
            assert previous_goal_location is not None
            final_boss_location.access_rule = previous_goal_location.can_reach
```